src/main/java/io/nats/client/api/BucketConfiguration.java

```diff
@@ -0,0 +1,292 @@
+// Copyright 2021 The NATS Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package io.nats.client.api;
+
+import java.time.Duration;
+
+import static io.nats.client.support.NatsKeyValueUtil.*;
+import static io.nats.client.support.Validator.*;
+
+/**
+ * The BucketConfiguration class contains the configuration for of a Key Value bucket.
+ */
+public class BucketConfiguration {
+    private final StreamConfiguration sc;
+    private final String name;
+
```

Comment:
The bucket configuration is really just stream configuration, but we want to present it as a bucket, not a stream, plus all settings are not relevant.